### ISSUE TITLE
Resuming training with Prodigy optimizer freezes LR to d0's value

### ISSUE THREAD

**Issue by gordigo** [kohya-ss/sd-scripts#857]
As title says, I've trained a SDXL LoRA with prodigy optimizer, I tried to resume using the exact same settings, but the LR flatlines at d0's default value of 1e-6 as shown here, I don't know if the issue is in Kohya or in the optimizer so I'm opening it here first

![imagen](https://github.com/kohya-ss/sd-scripts/assets/23042093/de51d680-00d4-478c-84e4-7783fe58ee2b)

This was the Accelerate command, I'm running under WSL2 Ubuntu if that helps

accelerate launch --num_cpu_threads_per_process=2 "./sdxl_train_network.py" --enable_bucket --min_bucket_reso=640 --max_bucket_reso=1700 --pretrained_model_name_or_path="/home/gordigo/sd_xl_base_1.0_0.9vae.safetensors" --train_data_dir="/mnt/f/kohya-dataset" --resolution="1024,1024" --output_dir="/mnt/f/output/" --logging_dir="/mnt/f/log" --network_alpha="32" --save_model_as=safetensors --network_module=networks.lora --network_dim=64 --network_weights="/mnt/f/data/Prodigy-LoRA.safetensors" --output_name="Prodigy-LoRA-Resumed --lr_scheduler_num_cycles="60" --no_half_vae --full_bf16 --learning_rate="1.0" --lr_scheduler="constant" --train_batch_size="6" --max_train_steps="2001" --save_every_n_epochs="1" --mixed_precision="bf16" --save_precision="bf16" --seed="807501" --caption_extension=".txt" --cache_latents --cache_latents_to_disk --optimizer_type="Prodigy" --optimizer_args decouple=True  weight_decay=0.01 d_coef=1 use_bias_correction=True  --max_data_loader_n_workers="0" --max_token_length=225 --caption_dropout_rate="0.05" --bucket_reso_steps=32 --min_snr_gamma=5 --flip_aug --shuffle_caption --gradient_checkpointing --xformers --noise_offset=0.0 --log_with wandb --wandb_api_key="*" --sample_sampler=euler_a --sample_prompts="/mnt/f/prompt.txt" --sample_every_n_steps="100"

Thank you so much for your hard work!

**Comment by gordigo** [comment]
Thank you very much, I'll do just that!
